### PR TITLE
chore(ocsf): add OCSF class for outputs

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

It's needed to modularize the code for the specific ocsf output.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/__main__.py [hunk 1]
```diff
@@ -309,12 +313,16 @@ def prowler():
...
-                close_json(
-                    global_provider.output_options.output_filename,
-                    global_provider.output_options.output_directory,
-                    mode,
+                filename = (
+                    f"{global_provider.output_options.output_directory}/"
+                    f"{global_provider.output_options.output_filename}{json_ocsf_file_suffix}"
                 )
-
+                json_finding = OCSF(
```
Comment (sergargar) [inline prowler/__main__.py:320]: Why not JSON_OCSF?

**Review by MrCloudSec — DISMISSED** [review]
LGTM!

File: prowler/__main__.py [hunk 2]
```diff
@@ -309,12 +313,16 @@ def prowler():
             # Close json file if exists
             # TODO: generate JSON here
             if "json" in mode:
```
Comment (jfagoagas) [inline prowler/__main__.py]:
```suggestion
            if "json-ocsf" in mode:
```

File: prowler/lib/outputs/ocsf/ocsf.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+    def transform(self, findings: List[Finding]) -> None:
+        """Transforms the findings into the OCSF format.
+
+        Args:
+            findings (List[Finding]): a list of Finding objects
+        """
+        try:
+            for finding_output in findings:
+                finding = deepcopy(finding_output)
```
Comment (jfagoagas) [inline prowler/lib/outputs/ocsf/ocsf.py]: I think you don't need this copy as far as you are not modifying the `finding` object.

File: prowler/lib/outputs/ocsf/ocsf.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+                        "categories": finding.categories,
+                        "depends_on": finding.depends_on,
+                        "related_to": finding.related_to,
+                        "notes": finding.notes,
+                        "compliance": finding.compliance,
+                    },
+                )
+
+                if finding.provider == "kubernetes":
+                    detection_finding.container = (
```
Comment (jfagoagas) [inline prowler/lib/outputs/ocsf/ocsf.py]: Please cover this case too.

File: tests/lib/outputs/json_ocsf/json_ocsf_test.py [hunk 5]
```diff
@@ -1,242 +0,0 @@
...
-
-        assert result == StatusID.Suppressed
-
-    # Returns StatusID.Other when muted is False and status is not "FAIL"
-    def test_other_when_status_pass_and_not_muted(self):
-        status = "PASS"
-        muted = False
-        result = get_finding_status_id(status, muted)
-
-        assert result == StatusID.Other
```
Comment (jfagoagas) [inline tests/lib/outputs/json_ocsf/json_ocsf_test.py:242]: Please move this tests to the new test class, you don't need to create a `ASFF` class just call the static methods with `ASFF.method(params)`

File: prowler/lib/outputs/file_descriptors.py
Comment (jfagoagas) [inline prowler/lib/outputs/file_descriptors.py:1]: Don't we need to add the `if` not to generate empty compliance files? Until this file is removed.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/__main__.py (hunk 2 above)
Comment (pedrooot) [reply prowler/__main__.py]: Done

File: tests/lib/outputs/json_ocsf/json_ocsf_test.py (hunk 5 above)
Comment (pedrooot) [reply tests/lib/outputs/json_ocsf/json_ocsf_test.py:242]: Done

File: prowler/lib/outputs/ocsf/ocsf.py (hunk 4 above)
Comment (pedrooot) [reply prowler/lib/outputs/ocsf/ocsf.py]: Ok, done

File: prowler/lib/outputs/ocsf/ocsf.py (hunk 3 above)
Comment (pedrooot) [reply prowler/lib/outputs/ocsf/ocsf.py]: True, done

File: prowler/__main__.py (hunk 2 above)
Comment (pedrooot) [reply prowler/__main__.py:315]: Done

File: prowler/lib/outputs/file_descriptors.py
Comment (pedrooot) [reply prowler/lib/outputs/file_descriptors.py:1]: I don't think that's needed

**Review by MrCloudSec — APPROVED** [review]
Re-reviewed all changes.

**Review by jfagoagas — APPROVED** [review]
👍

**Pull request merged**